libexec/apply_catalog.rb

```diff
@@ -50,6 +50,13 @@
   end
 
   env = Puppet.lookup(:environments).get('production').override_with(modulepath: [moduledir])
+  # Needed to ensure features are loaded
+  env.each_plugin_directory do |dir|
+    $LOAD_PATH << dir unless $LOAD_PATH.include?(dir)
```

Comment:
This would have broken providers with non-relative requires too right?

Reply:
Probably.

Reply:
Extended the tests to include require from puppet_x.

Reply:
But acceptance tests may not be working yet.

Reply:
Now passes.

Reply:
Tested with both Windows and Debian controllers.